systemvm/debian/opt/cloud/bin/cs/CsAddress.py

```diff
@@ -388,7 +388,7 @@ def fw_router(self):
             self.fw.append(["mangle", "",
                             "-A VPN_%s -j RETURN" % self.address['public_ip']])
             self.fw.append(["nat", "",
-                            "-A POSTROUTING -o eth2 -j SNAT --to-source %s" % self.address['public_ip']])
+                            "-A POSTROUTING -o %s -j SNAT --to-source %s" % (self.dev, self.address['public_ip'])])
```

Comment:
The attribute `self.dev` is already populated with the correct device name?

Reply:
Yes @rafaelweingartner, the logic is same what is being used in vpc routers, see - https://github.com/apache/cloudstack/blob/4.11/systemvm/debian/opt/cloud/bin/cs/CsAddress.py#L584

Reply:
Thanks for the explanation.

Reply:
Thanks for confirming that we indeed need to set the device/interface in the rule @ustcweizhou. I think it makes sense to have the default packets routed to the default public snat ip. I'll let others comment on it.

Reply:
Hi @ustcweizhou - in your scenario - does this not mean you end up SNATing on the wrong interface? So lets say you have eth2/VLAN10/10.100.1.10/24 and eth3/VLAN20/10.100.2.20/24. With your method that would SNAT to

- A POSTROUTING -o eth3 -j SNAT --to-source 10.100.1.10

So in other words the IP address would be right for the initial source NAT, but it would end up being translated on the wrong interface?

Reply:
@DagSonsteboSB yes, I think the outgoing IP for the vms without static nat should be the source nat IP, in your example 10.100.1.10

In my point of view, the rules applied on ethernet other than eth2 (for example eth3) are not needed at all, however it is not harmful to have them.

Reply:
Thanks all for your discussion and testing, I'll test @ustcweizhou 's fix now.

Reply:
You are right @ustcweizhou - just tested this. Routing happens *after SNAT* hence translating all non-static NATed traffic to the source IP address on eth2 works even if this means a SNAT rule against eth3, eth4, etc.